File: scientific_data_analysis_backend/app/schemas/task.py
"""
任务相关 Schema。
"""
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, ConfigDict

from app.models.enums import TaskStage, SuggestionStatus


class TaskCreateRequest(BaseModel):
    """创建任务请求。"""

    dataset_id: str


class TaskResponse(BaseModel):
    """任务响应。"""

    id: str
    dataset_id: str
    owner_id: str
    stage: TaskStage
    suggestion_status: SuggestionStatus
    active_version_id: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class TaskStatusResponse(BaseModel):
    """任务状态响应。"""

    task_id: str
    stage: TaskStage
    message: Optional[str] = None

File: frontend/src/types/task.ts
export type TaskStage =
  | 'uploading'
  | 'parsed'
  | 'profiling'
  | 'suggestion_pending'
  | 'processing'
  | 'analysis_ready'
  | 'visualization_ready';

export type SuggestionStatus = 'pending' | 'accepted' | 'rejected' | 'skipped';

export type SharePermission = 'view' | 'edit';

export interface Task {
  id: string;
  datasetId: string;
  ownerId: string;
  stage: TaskStage;
  suggestionStatus: SuggestionStatus;
  activeVersionId?: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface TaskStatus {
  taskId: string;
  stage: TaskStage;
  message?: string | null;
}

export interface SuggestionItem {
  id: string;
  taskId: string;
  cleaning: string[];
  statistics: string[];
  chartRecommendations: string[];
  notes: string[];
  status: SuggestionStatus;
}

export interface TaskShare {
  id: string;
  taskId: string;
  memberId: string;
  permission: SharePermission;
  expiresAt?: string | null;
}
